source/adapters/opencl/command_buffer.cpp

```diff
@@ -11,61 +11,120 @@
 #include "command_buffer.hpp"
 #include "common.hpp"
 
-/// Stub implementations of UR experimental feature command-buffers
-
 UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferCreateExp(
-    [[maybe_unused]] ur_context_handle_t hContext,
-    [[maybe_unused]] ur_device_handle_t hDevice,
+    ur_context_handle_t hContext, ur_device_handle_t hDevice,
     [[maybe_unused]] const ur_exp_command_buffer_desc_t *pCommandBufferDesc,
-    [[maybe_unused]] ur_exp_command_buffer_handle_t *phCommandBuffer) {
+    ur_exp_command_buffer_handle_t *phCommandBuffer) {
 
-  cl_adapter::die("Experimental Command-buffer feature is not "
-                  "implemented for OpenCL adapter.");
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+  ur_queue_handle_t Queue = nullptr;
+  ur_result_t Result = urQueueCreate(hContext, hDevice, nullptr, &Queue);
+  if (Result != UR_RESULT_SUCCESS) {
```

Comment:
Minor: There is a macro in the adapter called UR_RETURN_ON_FAILURE that could be used to simplify this error checks